lib/Dialect/AIE/IR/AIEDialect.cpp

```diff
@@ -429,6 +429,46 @@ xilinx::AIE::TileOp xilinx::AIE::ObjectFifoCreateOp::getProducerTileOp() {
   return cast<xilinx::AIE::TileOp>(getProducerTile().getDefiningOp());
 }
 
+// ObjectFifoLinkOp
+LogicalResult xilinx::AIE::ObjectFifoLinkOp::verify() {
+  ObjectFifoCreateOp fifoIn = getFifoIn().getDefiningOp<ObjectFifoCreateOp>();
+  AIEObjectFifoType fifoType = fifoIn.getType().cast<AIEObjectFifoType>();
+  MemRefType elemType = fifoType.getElementType().cast<MemRefType>();
+  int inputSize = (int)elemType.getShape()[0];
+
+  if (elemType.getShape().size() > 1)
+    return emitError("ObjectFifoLinkOp currently only supports objFifos with "
+                     "1-dimensional memrefs");
```

Comment:
I don't understand this restriction. Why does the object fifo link op care about the rank of the memref?

Reply:
Thank you for asking! The LinkOp shouldn't care. I removed the limitation and adjusted the code in the lowering.